community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachineSPI.java

```diff
@@ -178,4 +168,16 @@ public void terminate()
 
         };
     }
+
+    private static TransactionIdTracker newTransactionIdTracker( GraphDatabaseAPI db, AvailabilityGuard guard )
+    {
+        Supplier<TransactionIdStore> transactionIdStoreSupplier = db.getDependencyResolver().provideDependency( TransactionIdStore.class );
+        return new TransactionIdTracker( transactionIdStoreSupplier, guard );
+    }
+
+    private static TransactionalContextFactory newTransactionalContextFactory( GraphDatabaseAPI db )
+    {
+        GraphDatabaseQueryService queryService = db.getDependencyResolver().resolveDependency( GraphDatabaseQueryService.class );
```

Comment:
Since `GraphDatabaseQueryService` is the smaller interface which implements a subset of `GraphDatabaseAPI`, I wonder if it would make sense to switch this around so that we only expose  `GraphDatabaseQueryService` to Bolt and inject that as the dependency, if that is all we ever need?

Reply:
It makes sense to use `GraphDatabaseQueryService` because it is a smaller interface. However, it is instantiated after kernel extensions and can't be injected into `BoltKernelExtension`.

Reply:
I guess we could inject a `Supplier<GraphDatabaseQueryService>`, but it doesn't matter that much if we are soon going to replace it with some kind of database manager anyway.